internal/corazawaf/waf.go

```diff
@@ -380,5 +384,9 @@ func (w *WAF) Validate() error {
 		return errors.New("response body limit should be at most 1GB")
 	}
 
+	if w.ArgumentLimit <= 0 {
```

Comment:
Does `0` not make sense, prevent any argument?

Reply:
you mean that check the argument limit at internal/seclang/directives.go:directiveSecArgumentsLimit?

Reply:
I think @anuraaga means if `0` is a legit value or not cc @jptosso @fzipi @airween @dune73

Reply:
What does the question refer to? I mean do you want to know how ModSecurity (2 and 3) works (for the compatibility), or what would be the good behavior?

```
SecArgumentsLimit 0
```

In case of mod_security2:
```
ModSecurity: Invalid value for SecArgumentsLimit: 0
```

In case of libmodsecurity3:
* no error message
* based on the source code, 0 means 0 argument expected - see [here](https://github.com/SpiderLabs/ModSecurity/blob/da8782ce72f0950ab2ac408bc32e71a10f2d2a9f/src/transaction.cc#L396-L400)

Reply:
Can any of you think of a reasonable case where `0` is desirable? To me it does not make sense.

Reply:
Personally, I find it hard to imagine the need to set an unlimited value for this variable.

Reply:
One could use that to reject all arguments. But then that can also be done with a rule, so I don't see the benefit. The limit is clearly there to prevent resource exhaustion.